pkg/diffs/diffs.go

```diff
@@ -0,0 +1,66 @@
+package diffs
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+// Diffs holds the information where the differences can be found.
+type Diffs struct {
+	ConfigPath     string
+	JobConfigPath  string
+	DecorationPath string
+}
+
+// NewDiffs creates a new Diffs{}
+func NewDiffs(configPath, jobConfigPath, decorationPath string) *Diffs {
+	return &Diffs{
+		ConfigPath:     configPath,
+		JobConfigPath:  jobConfigPath,
+		DecorationPath: decorationPath,
+	}
+}
+
+// GetPresubmitsToExecute returns a mapping of repo to presubmits to execute.
+func (d *Diffs) GetPresubmitsToExecute() (map[string][]prowconfig.Presubmit, error) {
+	preSubmitsToExecuteMap := make(map[string][]prowconfig.Presubmit)
```

Comment:
Do we need to pollute the names with type information (the `Map` suffix)? It's rarely useful elsewhere but maybe I'm missing some Go folk knowledge.

Reply:
Also the reason why it won't crash, mentioned [#comment](https://github.com/openshift/ci-operator-prowgen/pull/52#discussion_r245669029)